lib/puppet/util/run_mode.rb

```diff
@@ -38,6 +38,10 @@ def log_dir
         "$vardir/log"
       end
 
+      def conf_dir
+        "$puppetdir/config"
+      end
+
```

Comment:
| A solution would be to use "#{puppet_dir}/config" instead of "$puppetdir/config"

The implementation `"#{puppet_dir}/config"` doesn't honor the explicit `--puppetdir=/foo/bar` when ran from source because it's just calling the `puppet_dir` method which will just return the hard-coded defaults - it won't interpolate my --puppetdir setting.

Trying to come up with a solution that works for both packages and source...